### PR TITLE
[Mellanox] Integrate HW-MGMT 7.0040.1008 Changes

### PR DESCRIPTION
**Pull request by vivekrnv**
## Patch List
* 0001-platform-mellanox-Introduce-support-for-rack-manager.patch : https://github.com/torvalds/linux/commit/f8dacbf7da2e
* 0002-platform-mellanox-Change-reset_pwr_converter_fail-at.patch : https://github.com/torvalds/linux/commit/488f0fca0db0
* 0003-platform-mellanox-Cosmetic-changes-rename-to-more-co.patch : https://github.com/torvalds/linux/commit/acc6ea304590
* 0004-platform-mellanox-Introduce-support-for-next-generat.patch : https://github.com/torvalds/linux/commit/fcf3790b9b63
* 0005-platform-mellanox-Introduce-support-of-new-Nvidia-L1.patch : https://github.com/torvalds/linux/commit/dd635e33b5c9
* 0006-platform-mellanox-Split-initialization-procedure.patch : https://github.com/torvalds/linux/commit/0170f616f496
* 0007-platform-mellanox-Split-logic-in-init-and-exit-flow.patch : https://github.com/torvalds/linux/commit/158cd8320776
* 0008-platform-mellanox-Extend-all-systems-with-I2C-notifi.patch : https://github.com/torvalds/linux/commit/233fd7e44cd7
* 0009-platform-mellanox-mlx-platform-Add-mux-selection-reg.patch : https://github.com/torvalds/linux/commit/cefdbc781566
* 0010-platform-mellanox-mlx-platform-Move-bus-shift-assign.patch : https://github.com/torvalds/linux/commit/50b823fdd357
* 0011-platform-mellanox-mlx-platform-Initialize-shift-vari.patch : https://github.com/torvalds/linux/commit/1a0009abfa78
* 0012-platform-mellanox-Fix-order-in-exit-flow.patch :
* 0013-platform-mellanox-mlx-platform-Fix-signals-polarity-.patch :
* 0014-platform-mellanox-mlx-platform-Modify-graceful-shutd.patch :
* 0015-platform-mellanox-Change-register-offset-addresses.patch :
* 0016-platform-mellanox-Add-new-attributes.patch :
* 0017-platform-mellanox-Add-field-upgrade-capability-regis.patch :
* 0018-platform-mellanox-Modify-reset-causes-description.patch :
* 0019-platform-mellanox-mlx-platform-Modify-health-and-pow.patch :
* 0020-platform-mellanox-mlx-platform-Add-reset-cause-attri.patch :
* 0021-platform-mellanox-mlx-platform-add-support-for-addit.patch :
* 0022-platform-mellanox-mlx-platform-Modify-power-off-call.patch :
* 0023-platform-mellanox-Cosmetic-changes.patch :
* 0024-platform-mellanox-mlx-platform-Add-reset-callback.patch :
* 0025-platform-mellanox-mlx-platform-Prepare-driver-to-all.patch :
* 0026-platform-mellanox-mlx-platform-Introduce-ACPI-init-f.patch :
* 0027-platform-mellanox-mlx-platform-Get-interrupt-line-th.patch :
* 0028-platform-mellanox-Add-initial-support-for-PCIe-based.patch :
* 0029-platform-mellanox-mlxreg-hotplug-Extend-condition-fo.patch :
* 0030-platform-mellanox-nvsw-sn2201-change-fans-i2c-busses.patch :
* 0032-platform_data-mlxreg-Add-field-with-mapped-resource-.patch : https://github.com/torvalds/linux/commit/26917eab144c
* 0033-i2c-mlxcpld-Allow-driver-to-run-on-ARM64-architectur.patch :
* 0034-i2c-mlxcpld-Add-support-for-extended-transaction-len.patch :
* 0035-i2c-mlxcpld-Support-PCIe-mapped-register-space.patch :
* 0039-mlxsw-reg-Limit-MTBR-register-payload-to-a-single-da.patch :
* 0040-mlxsw-core-Extend-allowed-list-of-external-cooling-d.patch :
* 0041-mlxsw-i2c-Utilize-standard-macros-for-dividing-buffe.patch :
* 0042-hwmon-mlxreg-fan-Add-support-for-new-flavour-of-capa.patch :
* 0043-hwmon-mlxreg-fan-Extend-number-of-supporetd-fans.patch :
* 0044-leds-mlxreg-Add-support-for-new-flavour-of-capabilit.patch :
* 0045-leds-mlxreg-Remove-code-for-amber-LED-colour.patch :
* 0046-Extend-driver-to-support-Infineon-Digital-Multi-phas.patch :
* 0047-dt-bindings-trivial-devices-Add-infineon-xdpe1a2g7.patch :
* 0048-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2891-c.patch :
* 0049-dt-bindings-trivial-devices-Add-mps-mp2891.patch :
* 0050-leds-mlxreg-Skip-setting-LED-color-during-initializa.patch :
* 0051-platform-mellanox-mlxreg-hotplug-Allow-more-flexible.patch : https://github.com/torvalds/linux/commit/26e118ea98cf
* 0052-i2c-mux-Add-register-map-based-mux-driver.patch :
* 0053-platform-mellanox-Add-support-for-dynamic-I2C-channe.patch :
* 0054-platform-mellanox-Introduce-support-for-switches-equ.patch :
* 0055-mellanox-Relocate-mlx-platform-driver.patch :
* 0056-Documentation-ABI-Add-new-attribute-for-mlxreg-io-sy.patch : https://github.com/torvalds/linux/commit/e7210563432a
* 0057-Documentation-ABI-Add-new-attribute-for-mlxreg-io-sy.patch : https://github.com/torvalds/linux/commit/e7210563432a
* 0085-hwmon-mlxreg-fan-Separate-methods-of-fan-setting-com.patch :
* 0087-platform-mellanox-indicate-deferred-I2C-bus-creation.patch :
* 0087-platform_data-mlxreg-Add-capability-bit-and-mask-fie.patch :
* 0088-platform-mellanox-mlxreg-hotplug-Add-support-for-new.patch :
* 0089-platform-mellanox-mlx-platform-Change-register-name.patch :
* 0090-platform-mellanox-mlx-platform-Add-support-for-new-X.patch :
* 0091-platform-mellanox-mlxreg-dpu-Add-initial-support-for.patch :
* 0092-platform-mellanox-Introduce-support-of-Nvidia-smart-.patch :
* 0093-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2855-c.patch :
* 0094-i2c-asf-Introduce-MCTP-support-over-ASF-controller.patch :
* 8000-mlxsw-Use-weak-reverse-dependencies-for-firmware-fla.patch :
* 8003-mlxsw-i2c-SONIC-ISSU-Prevent-transaction-execution-f.patch :
* 8004-mlxsw-minimal-Downstream-Ignore-error-reading-SPAD-r.patch :
* 8005-leds-leds-mlxreg-Downstream-Send-udev-event-from-led.patch :
* 8006-i2c-mlxcpld-Downstream-WA-to-avoid-error-for-SMBUS-r.patch :
* 8007-hwmon-mlxreg-fan-Downstream-Allow-fan-speed-setting-.patch :
* 8008-hwmon-emc2305-Downstream-Allow-fan-speed-setting-gra.patch :
* 8009-hwmon-mlxsw-Downstream-Allow-fan-speed-setting-granu.patch :
* 8010-mlxsw-i2c-Downstream-Add-retry-mechanism-for-failed-.patch :
* 8011-mlxsw-minimal-Downstream-Disable-ethtool-interface.patch :

### CODE REVIEW

File: patch/kconfig-inclusions [hunk 1]
```diff
@@ -35,12 +35,15 @@ CONFIG_SENSORS_MAX31790=m
...
 CONFIG_X86_AMD_PLATFORM_DEVICE=y
+CONFIG_MCTP=y
 CONFIG_DMI_SYSFS=y
 CONFIG_OF=y
 CONFIG_EEPROM_AT24=m
 CONFIG_AMD_XGBE=m
 CONFIG_IGB=m
 CONFIG_MLXSW_CORE=m
+CONFIG_USB_NET_DRIVERS=m
+CONFIG_USB_USBNET=m
```
Comment (saiarcot895) [inline patch/kconfig-inclusions]: Just to confirm, are these two Kconfigs actually needed?

Comment (saiarcot895) [reply patch/kconfig-inclusions]: Based on https://www.kernelconfig.io/config_usb_usbnet, it seems to be used more for older devices/hardware.

Comment (vivekkarri) [reply patch/kconfig-inclusions]: updated, removed the kconfig

**Comment by saiarcot895** [comment]
Can you remove the unused patch files?

**Comment by vivekrnv** [comment]
Updated

**Pull request merged**